### PR TITLE
Remove zero padding from `create_buffer_init`

### PR DESCRIPTION
**Pull request by grovesNL**
Follow-up to #870 (https://github.com/gfx-rs/wgpu-rs/pull/870#issuecomment-824333926)

### CODE REVIEW

File: src/util/device.rs [hunk 1]
```diff
@@ -56,10 +56,6 @@ impl DeviceExt for crate::Device {
         {
             let mut slice = buffer.slice(..).get_mapped_range_mut();
```
Comment (kvark) [inline src/util/device.rs:57]: maybe pass `unpadded_size` here and then use `slice` without brackets?

**Review by kvark — COMMENTED** [review]
let's squash and possibly make the proposed change

File: src/util/device.rs (hunk 1 above)
Comment (grovesNL) [reply src/util/device.rs]: Ah yeah I guess we don't even need `slice` here at all

**Review by kvark — APPROVED** [review]
Thank you!
bors r+

**Pull request merged**